### PR TITLE
Task 5

### PR DESCRIPTION
**Pull request by nobleaustine**
Task 5 completed

### CODE REVIEW

File: task_5/task_5.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+        
+    # downloading images within a range
+    def download_range(self, start: int, stop: int) -> list:
+        image_paths = []
+        for i in range(start, stop):
+            image_path = self.download_image(i)
+            image_paths.append(image_path)
+        return image_paths
+
+# Testing
```
Comment (FayazRahman) [inline task_5/task_5.py:76]: add test as a pytest in separate file. See https://docs.pytest.org/en/7.1.x/getting-started.html#create-your-first-test

File: task_5/task_5.py [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+            return self.download_image(key)
+        
+    # download an image provided the index
+    def download_image(self, index: int) -> str:
+        
+        # setting up url and image name
+        url = self.df["URL"][index]
+        URL = url.split('/')[-1]
+        extension = URL.split('?')[0]
+        ext = extension.split('.')[-1]
```
Comment (FayazRahman) [inline task_5/task_5.py:44]: Check if `os.path.splitext` will be enough here.

**Review by FayazRahman — APPROVED** [review]

**Review by FayazRahman — CHANGES_REQUESTED** [review]
Small changes needed

**Comment by nobleaustine** [comment]
used os.path.splitext and added separate test file

**Review by FayazRahman — CHANGES_REQUESTED** [review]
Rename `test_Downloader.py` to `test_downloader.py`

**Comment by nobleaustine** [comment]
updated the name to test_downloader.py

**Review by FayazRahman — APPROVED** [review]

**Comment by nobleaustine** [comment]
all issues resolved

**Pull request merged**